src/resources/views/crud/filters/date_range.blade.php

```diff
@@ -1,5 +1,18 @@
 {{-- Date Range Backpack CRUD filter --}}
 
+@php
+    $filterOptions = $filter->options['date_range_options'] ?? [];
+    $dateFormat = $filterOptions['format'] ?? config('backpack.base.default_date_format');
```

Comment:
Shouldn't ```format``` be inside a ```locale```? The way I see it, this filter should stick as much as possible to the [date_range field type](https://backpackforlaravel.com/docs/4.0/crud-fields#date-range) docs. For the field, you can do:

```php
[   // date_range
    'name' => ['start_date', 'end_date'], // db columns for start_date & end_date
    'label' => 'Event Date Range',
    'type' => 'date_range',
    // OPTIONALS
    'default' => ['2019-03-28 01:01', '2019-04-05 02:00'], // default values for start_date & end_date
    'date_range_options' => [
        // options sent to daterangepicker.js
        'timePicker' => true,
        'locale' => ['format' => 'DD/MM/YYYY HH:mm']
    ]
]
```

And I think the syntax should be exactly the same for the filter. Otherwise, having different syntaxes between these two would add to the learning curve.

Reply:
I see you've already used the same name for the options, ```date_range_options```, which is great [not the name, that's horrible ;-) but the fact that it's the same across similar fields and filters ]. That's why I'd expect what's inside this options array to work the same way too.

Reply:
Also notice it's [the same way that @axelzuzek tried to do it](https://github.com/Laravel-Backpack/CRUD/issues/2168#issue-510239148) when discovering the bug. Inside a "locale".

Reply:
Yes, i think about it. But when developing I thought, I am not going to use `locale` as primary options key, so i thought i would make "all" options available in JS locale, inside our `date_range_options`. 
I don't think it's very important to follow the structure of the JS because we only allow two parameters to be overriden, so keeping the same parameters names inside the `date_range_options` and writing them in the docs should be enough.

I don't expect someone in backpack when adding a daterange filter to go read the plugin documentation before reading our documentation. 

Last word is yours, let me know where to go :)

Best,
Pedro

Reply:
Hmmm I see your point... Not an easy decision... 

But I think what will be the best for the developer is for the syntax to be the same between fields and filters. That's the most intuitive - they use the same JS plugin, their syntax should be the same. 

I think we can construct a default configuration array, that uses the values we the user has given (if he's done so) otherwise fall back to our defaults. So basically 
```
$options = $filter->options['date_range_options'] ?? [];

(!isset($options['alwaysShowCalendars'])) ? $options['alwaysShowCalendars'] = true;
(!isset($options['autoUpdateInput'])) ? $options['autoUpdateInput'] = true;
// ...
```
He knows what he's doing if he overwrites them. Right?

What I don't know if we _can_ do is do the same for the attributes that use MomentJS. Namely:

```
 		        startDate: moment("{{ $start_date }}"),
 		        endDate: moment("{{ $end_date }}"),
```

So in JS we'd need to:
- json_decode the PHP ```$options``` array;
- add these two moment variables;

Not sure all this hassle is worth it :-) 
Let me think about it some more @pxpm .

Reply:
Hey @tabacitu . I give it a try hehe :) 

Updated the PR with your suggestions. Let me know what you think about it!

The main thing that should be added to docs is:

> If you specify a start/end date as a string (Can also be carbon but that is not problem.) you must ensure that the string is a mysql valid date YYYY-MM-DD or that you set FORMAT accordingly so date can be correctly parsed. 

All the rest i think i pretty much automated but i would like if you can give it a try :)

Best,
Pedro